test/functional/p2p_disconnect_ban.py

```diff
@@ -87,7 +88,7 @@ def run_test(self):
 
         # Move time forward by 3 seconds so the third ban has expired
         self.nodes[1].setmocktime(old_time + 3)
-        assert_equal(len(self.nodes[1].listbanned()), 4)
+        assert_equal(len(self.nodes[1].listbanned()), 5)
 
         self.log.info("Test ban_duration and time_remaining")
         for ban in self.nodes[1].listbanned():
```

Comment:
Recommend adding checks for the onion address in this loop.

Also recommend adding check for the onion address the after shutdown (technically after restart) here:
https://github.com/bitcoin/bitcoin/blob/605b1a84093f38a328940f96fcdc35866f5a7422/test/functional/p2p_disconnect_ban.py#L107-L111

Reply:
Nice idea, done!